src/cmt_decode_msgpack.c

```diff
@@ -800,6 +800,30 @@ static int unpack_meta_type(mpack_reader_t *reader, size_t index, void *context)
     return result;
 }
 
+static int unpack_agg_type(mpack_reader_t *reader, size_t index, void *context)
+{
+    uint64_t                           value;
+    int                                result;
+    struct cmt_msgpack_decode_context *decode_context;
+
+    if (NULL == reader ||
+        NULL == context) {
+        return CMT_DECODE_MSGPACK_INVALID_ARGUMENT_ERROR;
+    }
+
+    decode_context = (struct cmt_msgpack_decode_context *) context;
+
+    result = cmt_mpack_consume_uint_tag(reader, &value);
+
+    if (CMT_DECODE_MSGPACK_SUCCESS == result) {
+        decode_context->aggregation_type = value;
+
+        result = create_metric_instance(decode_context->map);
```

Comment:
The metric instance is created in `unpack_meta_type`, this line needs to be removed.
It's ok to save the value in the decoding context until it can be set later on though.

Reply:
done